enhancements/network/ovn-multi-external-gateway.md

```diff
@@ -0,0 +1,182 @@
+* `podGateways` allows defining the selector and configuration criteria for the gateway pods. This configuration contains the following fields:
+  * `network` references the multus network configuration that matches the same name in the pod that will be used for external traffic. If the field is empty or undefined, the pod's IP will be used instead. This matches the current behavior used with annotations. This field maps to the annotation `k8s.ovn.org/routing-network`.
+  * `bfdEnabled` determines if Bi-Directional Forwarding Detection (BFD) is supported in this network and follows the same behavior as with the annotation: it's enabled when set to true and disabled when false or omited in the manifest. It maps to the `k8s.ovn.org/bfd-enabled` annotation in the pod.
+  * `podSelector` implements a [set-based](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#set-based-requirement) label selector to filter the pods in the namespace that match this network configuration.
+
+The `status` structure in the CRD defines the following field:
+* `externalGatewayIPs` contains a slice of Ipv4 or IPv6 that are acting as external gateways. This field includes both the values from the `staticGateways` field as well as the IP values from the `podGateways`. It is updated by the controller when there are changes to the gateway pod IPs or to the static gateway values. This field exists in the status because the dynamic nature of the slice.
+
+The following yaml examples an instance of the CRD:
+
+```yaml
+kind: MultipleExternalGateway
+apiVersion: k8s.ovn.org/v1
+metadata:
+  name: multiple-external-gateways-configuration
+  namespace: default
+spec:
+  namespaces: ["default","stage"]
+  staticGateways:
+    externalIPs: ["9.0.0.1","9.0.0.2"]
+    bfdEnabled: true
+  podGateways:
+    - network: external-gateway
+      bfdEnabled: true
+      podSelector:
+        - matchLabels:
+          - external-gateway: "true"
+        - matchExpressions:
+          - {key: environment, operator: In, values: [stage]}
+    - network: vpn-network
+      podSelector:
+        - matchLabels:
+          - vpn-egress: "true"
+status:
+  externalGatewayIPs: ["9.0.0.1","9.0.0.2","10.10.1.2","10.10.1.5"]
+```
+
+### Risks and Mitigations
+N/A
+
+### Drawbacks
+N/A
+
+## Design Details
+
+
+### Test Plan
+
+* Unit test coverage for CR based to at least match use cases in existing annotation based test cases.
+* Unit test for both annotation and CR to include tests cases to validate expected behavior when running in OpenShift 4.13 and 4.14 as listed in the table above.
+
+### Graduation Criteria
+N/A
+
+#### Dev Preview -> Tech Preview
+This feature is GA in 4.13, no Dev or Tech previews are required. 
+
+
+#### Tech Preview -> GA
+This feature is GA in 4.13, no Dev or Tech previews are required.
+
+#### Removing a deprecated feature
+Deprecation of the annotation based logic to occur in OpenShift 4.14.
+
+### Upgrade / Downgrade Strategy
+N/A
```

Comment:
I think I mentioned this in another comment, but it would be good to outline how users currently using the annotations are expected to migrate to CRD? is it going to be manual?